test/Riok.Mapperly.Tests/Mapping/ObjectNestedPropertyTest.cs

```diff
@@ -0,0 +1,261 @@
+            "class A { public C Value { get; set; } }",
+            "class B { public string Id { get; set; } }",
+            "class C { public string Id { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                var target = new global::B();
+                target.Id = source.Value.Id;
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void DeeplyNestedPropertyWithMemberNameOfSource()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapNestedProperties(nameof(@A.Value.NestedValue))] partial B Map(A source);",
+            "class A { public C Value { get; set; } }",
+            "class B { public string Id { get; set; } }",
+            "class C { public D NestedValue { get; set; } }",
+            "class D { public string Id { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                var target = new global::B();
+                target.Id = source.Value.NestedValue.Id;
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void DeeplyNestedPropertyAsArrayWithMemberNameOfSource()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapNestedProperties(new [] {nameof(A.Value), nameof(C.NestedValue)})] partial B Map(A source);",
+            "class A { public C Value { get; set; } }",
+            "class B { public string Id { get; set; } }",
+            "class C { public D NestedValue { get; set; } }",
+            "class D { public string Id { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                var target = new global::B();
+                target.Id = source.Value.NestedValue.Id;
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void RootPropertiesShouldBePreferredOverNestedProperties()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapNestedProperties(nameof(A.Value))] partial B Map(A source);",
+            "class A { public string Id { get; set; } public C Value { get; set; } }",
+            "class B { public string Id { get; set; } }",
+            "class C { public string Id { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source, ignoreNestedMemberNotUsed)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                var target = new global::B();
+                target.Id = source.Id;
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void NestedPropertyWithSourcePathName()
```

Comment:
What is the difference between this test case and `NestedPropertyWithMemberNameOfSource`?